backend/src/test/java/com/redhat/cloud/notifications/routers/NotificationResourceTest.java

```diff
@@ -547,4 +547,16 @@ void testFindBehaviorGroupsByUnknownEventTypeId() {
                 .then()
                 .statusCode(404);
     }
+
+    @Test
+    void testBehaviorGroupsAffectedByRemovalOfUnknownEndpointId() {
+        Header identityHeader = initRbacMock("tenant", "user", FULL_ACCESS);
+        given()
+                .header(identityHeader)
+                .pathParam("endpointId", UUID.randomUUID())
+                .when()
+                .get("/notifications/behaviorGroups/affectedByRemovalOfEndpoint/{endpointId}")
+                .then()
+                .statusCode(404);
```

Comment:
This test fails because the change we talked is not in this PR yet. This is the method that needs to be changed:
https://github.com/RedHatInsights/notifications-backend/blob/5fb0654876102b626f06ec8a1575fc43f0b16f30/backend/src/main/java/com/redhat/cloud/notifications/db/repositories/BehaviorGroupRepository.java#L358-L369